From 2965547ee54fa62d198940ed024518bf4ad166dc Mon Sep 17 00:00:00 2001
From: Huang Shijie <b32955@freescale.com>
Date: Fri, 14 Mar 2014 15:13:36 +0800
Subject: [PATCH 0278/1543] ENGR00303533 mtd: fsl-quadspi: reset the module in
 the probe

commit b1acb92112ee8b847497140dd4a9a4a03eb7f5e2 from
git://git.freescale.com/imx/linux-2.6-imx.git

The uboot may run the QuadSpi controler with command:
    #sf probe

So we should reset the module in the probe.
This patch also clear the pending interrupts which arised by the uboot
code.

Signed-off-by: Huang Shijie <b32955@freescale.com>
---
 drivers/mtd/spi-nor/fsl-quadspi.c |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/drivers/mtd/spi-nor/fsl-quadspi.c b/drivers/mtd/spi-nor/fsl-quadspi.c
index 8d659a2..ad4335e 100644
--- a/drivers/mtd/spi-nor/fsl-quadspi.c
+++ b/drivers/mtd/spi-nor/fsl-quadspi.c
@@ -1,7 +1,7 @@
 /*
  * Freescale QuadSPI driver.
  *
- * Copyright (C) 2013 Freescale Semiconductor, Inc.
+ * Copyright (C) 2013-2014 Freescale Semiconductor, Inc.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -608,6 +608,11 @@ static int fsl_qspi_nor_setup(struct fsl_qspi *q)
 	if (ret)
 		return ret;
 
+	/* Reset the module */
+	writel(QUADSPI_MCR_SWRSTSD_MASK | QUADSPI_MCR_SWRSTHD_MASK,
+		base + QUADSPI_MCR);
+	udelay(1);
+
 	/* Init the LUT table. */
 	fsl_qspi_init_lut(q);
 
@@ -626,6 +631,7 @@ static int fsl_qspi_nor_setup(struct fsl_qspi *q)
 			base + QUADSPI_MCR);
 
 	/* enable the interrupt */
+	writel(0xffffffff, q->iobase + QUADSPI_FR);
 	writel(QUADSPI_RSER_TFIE, q->iobase + QUADSPI_RSER);
 
 	return 0;
-- 
1.7.5.4

